src/server/test/db/unitReadingsTests.js

```diff
@@ -0,0 +1,605 @@
+			expect(meterReadings.length).to.equal(365); // 365 days in a year
+			const aRow = meterReadings[0];
+			const rowWidth = moment.duration(aRow.end_timestamp.diff(aRow.start_timestamp));
+			expect(rowWidth.asDays()).to.equal(1);
+		});
+	});
+
+
+	// TODO modify so checks values too.
+
+	mocha.describe('Number of readings for underlying 15-minute data', () => {
+		let meter, graphicUnitId, conn;
+		const startDate = '2020-01-01 00:00:00';
+		// Do 61 days which is what is needed.
+		const endDate = '2020-03-02 00:00:00';
+
+		mocha.beforeEach(async function () {
+			// Extend timeout because a longer time with more data being created. The value is somewhat
+			// arbitrary and can be made larger if you get timeouts.
+			this.timeout(20000);
+			conn = testDB.getConnection();
+			// Insert the standard and special units and conversions. Really only need 1-2 but this is easy.
+			await Unit.insertStandardUnits(conn);
+			await Conversion.insertStandardConversions(conn);
+			await insertSpecialUnits(conn);
+			await insertSpecialConversions(conn);
+			await redoCik(conn);
+			// Make the meter be a kWh meter.
+			const meterUnitId = (await Unit.getByName('Electric_utility', conn)).id;
+			await new Meter(undefined, 'Meter', null, false, true, Meter.type.OTHER, 'CST', undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined,
+				undefined, undefined, undefined, undefined, meterUnitId, meterUnitId).insert(conn);
+			meter = await Meter.getByName('Meter', conn);
+			// Make the graphic unit be MegaJoules.
+			graphicUnitId = (await Unit.getByName('MJ', conn)).id;
+			const data = generateSineData(startDate, endDate, { timeStep: { minute: 15 } }).map(row => new Reading(meter.id, row[0], row[1], row[2]));
+			await Reading.insertAll(data, conn);
+			// Refresh daily and hourly reading views.
+			await Reading.refreshDailyReadings(conn);
+			await Reading.refreshHourlyReadings(conn);
+		});
+
+		mocha.it('Use Daily resolution when 61 days', async () => {
+			const start = moment(startDate);
+			const end = moment(startDate).clone();
+			const numDays = 61;
+			end.add(numDays, 'days');
+			const allReadings = await Reading.getMeterLineReadings([meter.id], graphicUnitId, start, end, conn);
+			// The min days is 61 for daily so should get daily points so 61.
+			expect(allReadings[meter.id].length).to.equal(numDays);
+		})
+
+		mocha.it('Use Daily resolution when 60 days', async () => {
+			const start = moment(startDate);
+			const end = moment(startDate).clone();
+			const numDays = 60;
+			end.add(numDays, 'days');
+			const allReadings = await Reading.getMeterLineReadings([meter.id], graphicUnitId, start, end, conn);
+			// The min days is 61 for daily so should get numDays days of hourly readings.
+			expect(allReadings[meter.id].length).to.equal(numDays * 24);
+		});
+
+		mocha.it('Us raw readings when 13 days', async () => {
```

Comment:
"Use" is misspelled here. The same for line 280.

Reply:
Thanks and done.